🐞 - Unable to install `@maskito/kit` without `--force` flag

### Which package(s) are the source of the bug?

@maskito/kit

### Description

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: admin-portal-frontend@0.0.0
npm ERR! Found: @maskito/core@0.2.0
npm ERR! node_modules/@maskito/core
npm ERR!   @maskito/core@"0.2.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @maskito/core@"^0.0.1" from @maskito/kit@0.2.0
npm ERR! node_modules/@maskito/kit
npm ERR!   @maskito/kit@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
```

I believe you need to update `@maskito/core` version in `@maskito/kit` `peerDependencies`

### Maskito version

0.2.0

### Which browsers have you used?

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

--- a/package.json
+++ b/package.json
@@ -157,7 +157,7 @@
     "standard-version": {
         "scripts": {
             "postchangelog": "ts-node ./scripts/postchangelog.ts",
-            "postbump": "node scripts/syncVersions.js && git add **/package.json"
+            "postbump": "ts-node ./scripts/postbump.ts"
         }
     }
 }

--- a/projects/angular/package.json
+++ b/projects/angular/package.json
@@ -34,7 +34,7 @@
     "peerDependencies": {
         "@angular/common": ">=12.0.0",
         "@angular/core": ">=12.0.0",
-        "@maskito/core": "^0.0.1",
+        "@maskito/core": "^0.2.0",
         "rxjs": ">=6.0.0"
     }
 }

--- a/projects/kit/package.json
+++ b/projects/kit/package.json
@@ -30,6 +30,6 @@
         "Nikita Barsukov <nikita.s.barsukov@gmail.com>"
     ],
     "peerDependencies": {
-        "@maskito/core": "^0.0.1"
+        "@maskito/core": "^0.2.0"
     }
 }

--- a/scripts/helpers/bump-maskito-deps.ts
+++ b/scripts/helpers/bump-maskito-deps.ts
@@ -1,0 +1,35 @@
+import {tuiIsString} from '@taiga-ui/cdk/utils';
+
+import {isMaskitoPackageName} from './is-maskito-package-name';
+
+export function bumpMaskitoDeps({
+    deps,
+    prevVersion,
+    newVersion,
+    isPeerDependency,
+    ignores,
+}: {
+    prevVersion: string;
+    newVersion: string;
+    deps: Record<string, Record<string, unknown> | string>;
+    isPeerDependency?: boolean;
+    ignores: string[];
+}): void {
+    const keys = Object.keys(deps).filter(key => isMaskitoPackageName(key, ignores));
+
+    for (const key of keys) {
+        if (tuiIsString(deps[key])) {
+            deps[key] = isPeerDependency
+                ? (deps[key] as string)?.replace(prevVersion, newVersion)
+                : `^${newVersion}`;
+        } else if (Object.prototype.hasOwnProperty.call(deps[key], 'requires')) {
+            bumpMaskitoDeps({
+                deps: (deps[key] as Record<string, Record<string, string>>).requires,
+                isPeerDependency,
+                ignores,
+                prevVersion,
+                newVersion,
+            });
+        }
+    }
+}

--- a/scripts/helpers/bump-maskito-deps.ts
+++ b/scripts/helpers/bump-maskito-deps.ts
@@ -1,5 +1,3 @@
-import {tuiIsString} from '@taiga-ui/cdk/utils';
-
 import {isMaskitoPackageName} from './is-maskito-package-name';
 
 export function bumpMaskitoDeps({
@@ -18,7 +16,7 @@ export function bumpMaskitoDeps({
     const keys = Object.keys(deps).filter(key => isMaskitoPackageName(key, ignores));
 
     for (const key of keys) {
-        if (tuiIsString(deps[key])) {
+        if (typeof deps[key] === 'string') {
             deps[key] = isPeerDependency
                 ? (deps[key] as string)?.replace(prevVersion, newVersion)
                 : `^${newVersion}`;

--- a/scripts/helpers/bump-maskito-version-in-package-json.ts
+++ b/scripts/helpers/bump-maskito-version-in-package-json.ts
@@ -1,0 +1,10 @@
+import {tuiIsString} from '@taiga-ui/cdk/utils';
+
+export function bumpMaskitoVersionInPackageJson(
+    packageJson: Record<string, unknown>,
+    version: string,
+): void {
+    if (`version` in packageJson && tuiIsString(packageJson[`version`])) {
+        packageJson[`version`] = version;
+    }
+}

--- a/scripts/helpers/bump-maskito-version-in-package-json.ts
+++ b/scripts/helpers/bump-maskito-version-in-package-json.ts
@@ -1,10 +1,8 @@
-import {tuiIsString} from '@taiga-ui/cdk/utils';
-
 export function bumpMaskitoVersionInPackageJson(
     packageJson: Record<string, unknown>,
     version: string,
 ): void {
-    if (`version` in packageJson && tuiIsString(packageJson[`version`])) {
+    if (`version` in packageJson && typeof packageJson[`version`] === 'string') {
         packageJson[`version`] = version;
     }
 }

--- a/scripts/helpers/is-maskito-package-name.ts
+++ b/scripts/helpers/is-maskito-package-name.ts
@@ -1,0 +1,5 @@
+export function isMaskitoPackageName(name: string = ``, ignores: string[]): boolean {
+    return (
+        (name?.startsWith(`@maskito/`) || name === `maskito`) && !ignores.includes(name)
+    );
+}

--- a/scripts/helpers/sync-versions.ts
+++ b/scripts/helpers/sync-versions.ts
@@ -1,0 +1,55 @@
+import {readFileSync, writeFileSync} from 'fs';
+import {glob} from 'glob';
+
+import {processLog, successLog} from './colored-log';
+import {updatePackageJsonStructure} from './update-package-json-structure';
+
+const INDENTATION = 4;
+
+export function syncVersions(
+    filesOrDirectories: string[],
+    newVersion: string,
+    ignores: string[] = [],
+): void {
+    const patterns = filesOrDirectories.map(pattern =>
+        pattern.endsWith(`.json`)
+            ? pattern
+            : `${pattern}/**/*(package.json|package-lock.json)`,
+    );
+
+    const files = patterns
+        .map(pattern => glob.sync(pattern, {ignore: `**/node_modules/**`}))
+        .flatMap(files => files)
+        .filter(file => !file.includes(`node_modules`));
+
+    for (const file of files) {
+        const originalJSON = `${JSON.stringify(
+            JSON.parse(readFileSync(file).toString()),
+            null,
+            INDENTATION,
+        )}`;
+        const packageJson = JSON.parse(originalJSON);
+        const prevVersion = packageJson.version;
+
+        if (!prevVersion) {
+            continue;
+        }
+
+        updatePackageJsonStructure({
+            isPackageLockFile: file.endsWith(`-lock.json`),
+            packageJson,
+            prevVersion,
+            newVersion,
+            ignores,
+        });
+
+        const updatedJSON = `${JSON.stringify(packageJson, null, INDENTATION)}`;
+
+        if (originalJSON !== updatedJSON) {
+            writeFileSync(file, `${updatedJSON}\n`);
+            successLog(`[synchronized]: ${file}`);
+        } else {
+            processLog(`[no changes]: ${file}`);
+        }
+    }
+}

--- a/scripts/helpers/update-package-json-structure.ts
+++ b/scripts/helpers/update-package-json-structure.ts
@@ -1,8 +1,12 @@
-import {tuiIsObject, tuiIsString} from '@taiga-ui/cdk/utils';
-
 import {bumpMaskitoDeps} from './bump-maskito-deps';
 import {bumpMaskitoVersionInPackageJson} from './bump-maskito-version-in-package-json';
 import {isMaskitoPackageName} from './is-maskito-package-name';
+
+function isObject<T extends Record<string, any>>(
+    value: unknown,
+): value is NonNullable<T> {
+    return typeof value === `object` && !!value;
+}
 
 interface UpdatePackageJsonOptions {
     packageJson: Record<string, Record<string, any> | string>;
@@ -21,15 +25,15 @@ export function updatePackageJsonStructure({
 }: UpdatePackageJsonOptions): void {
     const {name, dependencies, peerDependencies, devDependencies, packages} = packageJson;
 
-    if (tuiIsString(name) && isMaskitoPackageName(name, ignores)) {
+    if (typeof name === 'string' && isMaskitoPackageName(name, ignores)) {
         bumpMaskitoVersionInPackageJson(packageJson, newVersion);
     }
 
-    if (tuiIsObject(dependencies)) {
+    if (isObject(dependencies)) {
         bumpMaskitoDeps({deps: dependencies, prevVersion, newVersion, ignores});
     }
 
-    if (tuiIsObject(peerDependencies)) {
+    if (isObject(peerDependencies)) {
         bumpMaskitoDeps({
             deps: peerDependencies,
             prevVersion,
@@ -39,11 +43,11 @@ export function updatePackageJsonStructure({
         });
     }
 
-    if (tuiIsObject(devDependencies)) {
+    if (isObject(devDependencies)) {
         bumpMaskitoDeps({deps: devDependencies, prevVersion, newVersion, ignores});
     }
 
-    if (isPackageLockFile && tuiIsObject(packages)) {
+    if (isPackageLockFile && isObject(packages)) {
         for (const packageLockJson of Object.values(packages)) {
             if (!isMaskitoPackageName(packageLockJson?.name, ignores)) {
                 continue;

--- a/scripts/helpers/update-package-json-structure.ts
+++ b/scripts/helpers/update-package-json-structure.ts
@@ -1,0 +1,61 @@
+import {tuiIsObject, tuiIsString} from '@taiga-ui/cdk/utils';
+
+import {bumpMaskitoDeps} from './bump-maskito-deps';
+import {bumpMaskitoVersionInPackageJson} from './bump-maskito-version-in-package-json';
+import {isMaskitoPackageName} from './is-maskito-package-name';
+
+interface UpdatePackageJsonOptions {
+    packageJson: Record<string, Record<string, any> | string>;
+    prevVersion: string;
+    newVersion: string;
+    isPackageLockFile: boolean;
+    ignores: string[];
+}
+
+export function updatePackageJsonStructure({
+    packageJson,
+    isPackageLockFile,
+    ignores,
+    prevVersion,
+    newVersion,
+}: UpdatePackageJsonOptions): void {
+    const {name, dependencies, peerDependencies, devDependencies, packages} = packageJson;
+
+    if (tuiIsString(name) && isMaskitoPackageName(name, ignores)) {
+        bumpMaskitoVersionInPackageJson(packageJson, newVersion);
+    }
+
+    if (tuiIsObject(dependencies)) {
+        bumpMaskitoDeps({deps: dependencies, prevVersion, newVersion, ignores});
+    }
+
+    if (tuiIsObject(peerDependencies)) {
+        bumpMaskitoDeps({
+            deps: peerDependencies,
+            prevVersion,
+            newVersion,
+            isPeerDependency: true,
+            ignores,
+        });
+    }
+
+    if (tuiIsObject(devDependencies)) {
+        bumpMaskitoDeps({deps: devDependencies, prevVersion, newVersion, ignores});
+    }
+
+    if (isPackageLockFile && tuiIsObject(packages)) {
+        for (const packageLockJson of Object.values(packages)) {
+            if (!isMaskitoPackageName(packageLockJson?.name, ignores)) {
+                continue;
+            }
+
+            updatePackageJsonStructure({
+                packageJson: packageLockJson,
+                prevVersion,
+                newVersion,
+                isPackageLockFile: true,
+                ignores,
+            });
+        }
+    }
+}

--- a/scripts/postbump.ts
+++ b/scripts/postbump.ts
@@ -1,0 +1,8 @@
+import {version} from '../package.json';
+import {execute} from './helpers/execute';
+import {syncVersions} from './helpers/sync-versions';
+
+(function main(): void {
+    syncVersions([`./projects`, `./package-lock.json`], version);
+    execute(`git add .`);
+})();

--- a/tsconfig.base.json
+++ b/tsconfig.base.json
@@ -30,6 +30,7 @@
         "noImplicitReturns": true,
         "noUnusedParameters": true,
         "noUnusedLocals": true,
+        "resolveJsonModule": true,
         "target": "es2015",
         "typeRoots": ["node_modules/@types"],
         "lib": ["es2017", "es2018.asynciterable", "dom"],

